web/src/stream_topic_history_util.ts

```diff
@@ -33,9 +39,13 @@ export function get_server_history(stream_id: number, on_success: () => void): v
             const server_history = data.topics;
             stream_topic_history.add_history(stream_id, server_history);
             stream_topic_history.remove_request_pending_for(stream_id);
-            on_success();
+            for (const callback of pending_on_success_callbacks.get(stream_id)!) {
+                callback();
+            }
+            pending_on_success_callbacks.delete(stream_id);
         },
         error() {
+            pending_on_success_callbacks.delete(stream_id);
```

Comment:
This isn't a regression, but I'm not sure it's most correct to delete these... at the same time, more fancy error handling is probably a pain, so I think it's OK to have this as-is, or maybe add a brief `TODO` comment noting that it's possible some sort of retry logic would be appropriate.

Reply:
Added a comment:
```diff
        error() {
+           // TODO: Implement some sort of retry logic,
+           // before giving up on the first failure.
            pending_on_success_callbacks.delete(stream_id);
            stream_topic_history.remove_request_pending_for(stream_id);
        },
```